### PR TITLE
add `region` api to topo server, adding preference logic to discoverygateway

### PR DESCRIPTION
**Pull request by inexplicable**
…po server, updated `TabletStatsCache` and `discoverygateway` to use the region info

### CODE REVIEW

File: go/vt/vtgate/gateway/discoverygateway.go [hunk 1]
```diff
@@ -121,6 +125,24 @@ func createDiscoveryGateway(hc discovery.HealthCheck, topoServer topo.Server, se
...
+func buildCellsToRegions(topoServer topo.Server, cells []string) map[string]string {
+
+	cellsToRegions := make(map[string]string)
+
+	for _, c := range cells {
+		if c == "" {
+			continue
+		}
+		println(c)
+		region, error := topoServer.GetRegionByCell(c)
```
Comment (rafael) [inline go/vt/vtgate/gateway/discoverygateway.go]: nit - maybe:
```
if region, error := topoServer.GetRegionByCell(c); !error {
  cellsToRegions[c] = region
}
```

File: go/vt/discovery/tablet_stats_cache.go [hunk 2]
```diff
@@ -66,7 +70,7 @@ type tabletStatsCacheEntry struct {
 // SetListener with sendDownEvents=true, as we need these events
 // to maintain the integrity of our cache.
 func NewTabletStatsCache(hc HealthCheck, cell string) *TabletStatsCache {
```
[the remark this answers is not in the document]
Comment (inexplicable) [reply go/vt/discovery/tablet_stats_cache.go]: yep, it's being worked on, problem is, there're many places like `legacy_split_clone`, `topo_util`, `binlog_players` in which i'm not sure if:
1. topo server is available
2. cells to regions is required (whether the nature of the work needs to see cross cell tablets at all)

it seems to me that passing an explicit empty `map[string]string` would effectively keep a backward compatible behavior (see only in cell tablets), is that good enough?

[the remark this answers is not in the document]
Comment (rafael) [reply go/vt/discovery/tablet_stats_cache.go]: Yeah I think that makes sense!

File: proto/topodata.proto [hunk 3]
```diff
@@ -323,4 +323,7 @@ message CellInfo {
   // Root is the path to store data in. It is only used when talking
   // to server_address.
   string root = 2;
+
```
Comment (inexplicable) [inline proto/topodata.proto]: add `region` to `CellInfo`

**Pull request merged**